src/core/query.ts

```diff
@@ -460,28 +463,7 @@ export class Query<
           this.optionalRemove()
         }
       },
-      onError: (error: TError | { silent?: boolean }) => {
-        // Optimistically update state if needed
-        if (!(isCancelledError(error) && error.silent)) {
-          this.dispatch({
-            type: 'error',
-            error: error as TError,
-          })
-        }
-
-        if (!isCancelledError(error)) {
-          // Notify cache callback
-          this.cache.config.onError?.(error, this as Query<any, any, any, any>)
-
-          // Log error
-          getLogger().error(error)
-        }
-
-        // Remove query after fetching if cache time is 0
-        if (this.cacheTime === 0) {
-          this.optionalRemove()
-        }
-      },
+      onError: this.onError,
```

Comment:
it doesn't have to live on the `this` scope, I think that is too wide. Also binding comes with an overhead in bundle size I believe. It was an inline function before, so I think it can just be:

`const onError: (error: TError | { silent?: boolean }) => ...`

defined right above `this.retryer = new Retryer({`